src/providers/dataProvider/resource-callbacks/DestructionLifeCycle.ts

```diff
@@ -0,0 +1,27 @@
+import { type CreateResult, type ResourceCallbacks } from 'ra-core'
+
+import { withCreatedByAt, type AuditFunctionType } from '../dataprovider-utils'
+import { AuditType } from '../../../utils/activity-types'
+import { R_DESTRUCTION } from '../../../constants'
+
+export default (audit: AuditFunctionType): ResourceCallbacks<any> => ({
+  resource: R_DESTRUCTION,
+  beforeCreate: async (record: CreateResult<Destruction>) => {
```

Comment:
@salony75way  - the type of `record` for `beforeCreate` should be `CreateParams`

Reply:
Aah, we have this wrong across a lot of RCO. Will log separate issue.